internal/provider/resource_ko_resolve.go

```diff
@@ -0,0 +1,263 @@
+				Default:     "spdx",
+				Optional:    true,
+				Type:        schema.TypeString,
+				ForceNew:    true,
+			},
+			BaseImageKey: {
+				Description: "",
+				Default:     defaultBaseImage,
+				Optional:    true,
+				Type:        schema.TypeString,
+				ForceNew:    true,
+			},
+			TagsKey: {
+				Description: "Which tags to use for the produced image instead of the default 'latest' tag ",
+				Optional:    true,
+				Type:        schema.TypeList,
+				Elem:        &schema.Schema{Type: schema.TypeString},
+				ForceNew:    true,
+			},
+
+			// Computed fields
+			ManifestsKey: {
+				Description: "A list of resolved manifests in a 'yamldecode'able format. Note that whitespaces and nil docs will be stripped from these results.",
+				Type:        schema.TypeList,
+				Elem:        &schema.Schema{Type: schema.TypeString},
+				Computed:    true,
+			},
+		},
+	}
+}
+
+type Resolver interface {
+	Resolve(ctx context.Context) (*Resolved, error)
+}
+
+type Resolved struct {
+	Id        string
+	Manifests []string
+}
+
+type resolver struct {
+	bo *options.BuildOptions
+	po *options.PublishOptions
+	fo *options.FilenameOptions
+	so *options.SelectorOptions
+}
+
+func NewResolver(d *schema.ResourceData, meta interface{}) (*resolver, error) {
+	opts, err := NewProviderOpts(meta)
+	if err != nil {
+		return nil, err
+	}
+
+	r := &resolver{
+		bo: opts.bo,
+		po: opts.po,
+		fo: &options.FilenameOptions{},
+		so: &options.SelectorOptions{},
+	}
+
+	if p, ok := d.Get(BaseImageKey).(string); ok {
+		r.bo.BaseImage = p
+	}
+
+	if p, ok := d.Get(TagsKey).([]interface{}); ok {
+		if len(p) == 0 {
+			r.po.Tags = []string{"latest"}
+		} else {
+			r.po.Tags = StringSlice(p)
+		}
+	}
+
+	if p, ok := d.Get(PushKey).(bool); ok {
+		r.po.Push = p
+	}
+
+	if p, ok := d.Get(FilenamesKey).([]interface{}); ok {
+		r.fo.Filenames = StringSlice(p)
+	}
+
+	if p, ok := d.Get(RecursiveKey).(bool); ok {
+		r.fo.Recursive = p
+	}
+
+	if p, ok := d.Get(SelectorKey).(string); ok {
+		r.so.Selector = p
+	}
+
+	return r, nil
+}
+
+func (r *resolver) Resolve(ctx context.Context) (*Resolved, error) {
+	builder, err := commands.NewBuilder(ctx, r.bo)
+	if err != nil {
+		return nil, err
+	}
+
+	cacheBuilder, err := build.NewCaching(builder)
+	if err != nil {
+		return nil, err
+	}
+
+	publisher, err := commands.NewPublisher(r.po)
+	if err != nil {
+		return nil, err
+	}
+	defer publisher.Close()
+
+	var resolveBuf bytes.Buffer
+	w := &nopWriteCloser{Writer: bufio.NewWriter(&resolveBuf)}
+
+	if err := commands.ResolveFilesToWriter(ctx, cacheBuilder, publisher, r.fo, r.so, w); err != nil {
+		return nil, err
+	}
+
+	if err := w.Flush(); err != nil {
+		return nil, err
+	}
+
+	// Split the dump of multi-doc yaml back into their individual nodes
+	// NOTE: Don't use a strings.Split to ensure we filter out any null docs
+	manifests := []string{}
```

Comment:
Ahhh I see, we split it back out after passing it through `ResolveFilesToWriter`. Is this because downstream stuff like a k8s terraform provider expects individual single-resource manifests, and won't accept a single big `---`-delimited YAML dump?

Reply:
afaik there are 2 limitations with the dump everything approach:

1. `yamldecode` doesn't support multidoc yaml
2. the `kubernetes_manifest` resource is restricted to a single k8s resource per tf resource

so you're correct! the format of `manifests` is specifically to address the intended use of `ko_resolve`: "easily" pass it into `kubernetes_manifest`'s